src/export/ExportMP3.cpp

```diff
@@ -1859,7 +1859,17 @@ ProgressResult ExportMP3::Export(AudacityProject *project,
    // Verify sample rate
    if (!make_iterator_range( sampRates ).contains( rate ) ||
       (rate < lowrate) || (rate > highrate)) {
-      rate = AskResample(bitrate, rate, lowrate, highrate);
+       if (project->mBatchMode) {
+           if (bitrate > 32) {
+               rate = 48000;
+           }
+           else if (bitrate <= 32) {
+               rate = 240000;
```

Comment:
You meant 24000?

Reply:
Is the intention to default to `highrate` appropriate for the bit rate?  Then say so in a comment.

Reply:
Perhaps clear without resorting to comments like this?
```
   // Verify sample rate
   if (!make_iterator_range( sampRates ).contains( rate ) ||
      (rate < lowrate) || (rate > highrate)) {
         if (project->mBatchMode) {
            if (rate == 0) {  // Maybe possible from macro command?
               return ProgressResult::Cancelled;
            }
            else if (rate < lowrate) {
               rate = lowrate;
            }
            else {
               rate = highrate;
            }
         }
         else {
            rate = AskResample(bitrate, rate, lowrate, highrate);
            if (rate == 0) {
               return ProgressResult::Cancelled;
         }
      }
   }

```

Reply:
Thinking about it a bit more, I think it's possible to set an invalid MP3 bitrate via a macro, so to cover all possibilities we should ensure that macros also handle any *illegal* bitrate and not just rate == 0.

Reply:
Yes it is possible for a macro to set an invalid sample rate, so something like...
```
   // Verify sample rate
   if (!make_iterator_range( sampRates ).contains( rate ) ||
      (rate < lowrate) || (rate > highrate)) {
         if (project->mBatchMode) {
            // Force valid sample rate in macros.
            if (!make_iterator_range( sampRates ).contains( rate )) {
               auto const bestRateIt = std::lower_bound(sampRates.begin(),
                                                         sampRates.end(), rate);
               rate = (bestRateIt == sampRates.end()) ? highrate : *bestRateIt;
               // wxLogDebug("Best Rate: %d", rate);
            }
            if (rate < lowrate) {
               rate = lowrate;
            }
            else if (rate > highrate) {
               rate = highrate;
            }
         }
         else {
            rate = AskResample(bitrate, rate, lowrate, highrate);
            if (rate == 0) {
               return ProgressResult::Cancelled;
         }
      }
   }

```